src/main/java/de/tum/in/www1/artemis/web/rest/BonusResource.java

```diff
@@ -0,0 +1,253 @@
+        log.debug("REST request to create a bonus source for exam: {}", examId);
+        if (bonus.getId() != null) {
+            throw new BadRequestAlertException("A new bonus source cannot already have an ID", ENTITY_NAME, "idexists");
+        }
+
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+
+        GradingScale sourceGradingScaleFromDb = gradingScaleRepository.findById(bonus.getSourceGradingScale().getId()).orElseThrow();
+        bonus.setSourceGradingScale(sourceGradingScaleFromDb);
+        checkIsAtLeastInstructorForGradingScaleCourse(sourceGradingScaleFromDb);
+
+        GradingScale bonusToGradingScale = gradingScaleRepository.findWithEagerBonusFromByExamId(examId).orElseThrow();
+        if (bonusRepository.existsByBonusToGradingScaleId(bonusToGradingScale.getId())) {
+            // In the future, when multiple bonuses per bonusToGradingScale is supported this check should be removed.
+            throw new BadRequestAlertException("A bonus is already created for this bonusToGradingScale", ENTITY_NAME, "idexists");
+        }
+
+        bonusToGradingScale.addBonusFrom(bonus);
+        bonusToGradingScale.setBonusStrategy(bonus.getBonusStrategy());
+
+        Bonus savedBonus = bonusService.saveBonus(bonus, true);
+        gradingScaleRepository.save(bonusToGradingScale);
+
+        filterBonusForResponse(savedBonus);
+        return ResponseEntity.created(new URI("/api/courses/" + courseId + "/exams/" + examId + "/bonus/" + savedBonus.getId()))
+                .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, "")).body(savedBonus);
+    }
+
+    /**
+     * Sets redundant fields to null to save bandwidth and prevent circular dependencies.
+     * <p>
+     * Warning: Modifies the input argument.
+     *
+     * @param bonus that will be modified
+     */
+    private void filterBonusForResponse(Bonus bonus) {
+        if (bonus == null) {
+            return;
+        }
+
+        GradingScale bonusTo = bonus.getBonusToGradingScale();
+        if (bonusTo != null) {
+            // This line breaks the circular dependency between savedBonus and bonusToGradingScale
+            bonusTo.setBonusFrom(null);
+            bonusTo.setGradeSteps(null);
+        }
+
+        GradingScale source = bonus.getSourceGradingScale();
+        if (source != null) {
+            source.setGradeSteps(null);
+        }
+    }
+
+    /**
+     * PUT /courses/{courseId}/bonus : Update bonus source for course
+     *
+     * @param bonus the bonus source which will be updated
+     * @return ResponseEntity with status 200 (Ok) with body the newly updated bonus source if it is correctly formatted and 400 (Bad request) otherwise
+     */
+    @PutMapping("/bonus")
```

Comment:
inconsistent
please add courses/{courseId} or exams/{examId} before and definitely make it consistent with the other URLs

Reply:
Url is standardized by adding `courseId` and `examId`.